sway/config/output.c

```diff
@@ -277,17 +277,49 @@ void apply_output_config(struct output_config *oc, struct sway_output *output) {
 }
 
 static struct output_config *get_output_config(char *name, char *identifier) {
+	struct output_config *oc_name = NULL;
 	int i = list_seq_find(config->output_configs, output_name_cmp, name);
 	if (i >= 0) {
-		return config->output_configs->items[i];
+		oc_name = config->output_configs->items[i];
 	}
 
+	struct output_config *oc_id = NULL;
 	i = list_seq_find(config->output_configs, output_name_cmp, identifier);
 	if (i >= 0) {
-		return config->output_configs->items[i];
+		oc_id = config->output_configs->items[i];
 	}
 
-	return NULL;
+	struct output_config *result = NULL;
+	if (oc_name && oc_id) {
+		// Generate a config named `<identifier> on <name>` which contains a
+		// merged copy of the identifier on name. This will make sure that both
+		// identifier and name configs are respected, with identifier getting
+		// priority
+		size_t length = strlen(identifier) + strlen(" on ") + strlen(name) + 1;
+		char *temp = malloc(length);
+		snprintf(temp, length, "%s on %s", identifier, name);
+
+		result = new_output_config(temp);
+		merge_output_config(result, oc_name);
+		merge_output_config(result, oc_id);
+
+		wlr_log(WLR_DEBUG, "Generated output config for %s on %s (enabled: %d)"
+			" (%dx%d@%fHz position %d,%d scale %f transform %d) (bg %s %s)"
+			" (dpms %d)", name, identifier, result->enabled, result->width,
+			result->height, result->refresh_rate, result->x, result->y,
+			result->scale, result->transform, result->background,
+			result->background_option, result->dpms_state);
+	} else if (oc_name) {
+		// No identifier config, just return a copy of the name config
+		result = new_output_config(strdup(name));
```

Comment:
No need to `strdup` here -- `new_output_config` takes a `const char *`.

Reply:
Fixed